### ISSUE TITLE
break-before:left/right at beginning of a document does not work properly

### ISSUE THREAD

**Issue by MurakamiShinyu** [vivliostyle/vivliostyle.js#666]
CSS Paged Media spec https://drafts.csswg.org/css-page-3/#progression says:

> … To explicitly force a document to begin printing on a left or right page, authors can specify a 'break-before' value that that propagates a page break to the root. [CSS-BREAK-3] The UA must suppress the first (empty) page(s) in this case (and the `:first` pseudo-class matches the first printed page).
> 
> EXAMPLE 1
> …
> 
>     html { break-before: left }
> 
> For an HTML document, the above style rule will cause the first page of the document to print on a ':left' page, regardless of the page progression.

The current behavior of Vivliostyle.js differs from this specification in some points.

I made test files:

- [html-right.html](https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html)  -- html { break-before: right }
  - Vivliostyle: https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html
    - seems OK
- [html-left.html](https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-left.html)  -- html { break-before: left }
  - Vivliostyle: https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-left.html
    - BUG: `html { break-before: left }` does not work
- [body-right.html](https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html)   -- body { break-before: right }
  - Vivliostyle: https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html
    - seems OK
- [body-left.html](https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-left.html)  -- body { break-before: left }
  - Vivliostyle: https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-left.html
    - BUG: `body { break-before: left }` works but,
      - differs from the spec "The UA must suppress the first (empty) page(s) in this case (and the :first pseudo-class matches the first printed page)."
      - The first (empty) page has page-margin box content
- body-right.html + html-right.html
  - Vivliostyle: https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html&src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html
    - BUG: `html { break-before: right }` on the second document does not work
- html-right.html + body-right.html
  - Vivliostyle: https://vivliostyle.org/viewer/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html&src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html
    - BUG: `body { break-before: right }` on the second document works but,
      - The generated blank page between two documents has page-margin box content ("body-right") defined for the second document (body-right.html). This is incorrect because the document with `break-before: right` should begin on a right page, and the generated blank page should belong to the previous document.

In summary, there are the following problems:
- break-before: left/right on the root (`html`) element does not work
- break-before: left/right on the `body` element generates blank page even on the first document
- the blank page generated by the break-before: left/right on the `body` element of the Nth document in a multi-document publication (like EPUB) has page-margin box content defined for the Nth document. (not for the (N-1)th document).

**Comment by MurakamiShinyu** [comment]
#694 Fixed the problems:

### Fixed: break-before: left/right(verso/recto) on the root (html) element does not work

Test with the version the bug is found:
- https://vivliostyle.github.io/viewer/v2.4.2/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-left.html
- https://vivliostyle.github.io/viewer/v2.4.2/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html&src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html

Test with the latest canary version:
- https://vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-left.html
- https://vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html&src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html

Note: Now the first page (normally recto, i.e., right in LTR page progression direction) can be a verso (left in LTR) page.

### Fixed: break-before: left/right(verso/recto) on the body element generates blank page even on the first document

Test with the version the bug is found:
- https://vivliostyle.github.io/viewer/v2.4.2/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-left.html

Test with the latest canary version:
- https://vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-left.html

### Fixed: the blank page generated by the break-before: left/right(verso/recto) on the body element of the Nth document in a multi-document publication (like EPUB) has page-margin box content defined for the Nth document. (not for the (N-1)th document).

Test with the version the bug is found:
- https://vivliostyle.github.io/viewer/v2.4.2/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html&src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html

Test with the latest canary version:
- https://vivliostyle.vercel.app/#src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/html-right.html&src=https://gist.githubusercontent.com/MurakamiShinyu/1c5fc34daf20c52ec289b7e2084b3620/raw/body-right.html

Note: with this fix, the blank page between two documents is now completely blank (no page-margin boxes). I think this is the expected behavior.
